### PR TITLE
Add s390x support

### PR DESCRIPTION
**Pull request by raulcabello**
Signed-off-by: raulcabello <raul.cabello@suse.com>

#### Proposed Changes ####

Add support for s390x

#### Further Comments ####

- No airgap for s390x
- Move manifest step to a new drone pipeline, so it also add s390x image
- Bump versions because new version are available on s390x
- Remove socat as it was dropped by containerd 1.4

### CODE REVIEW

**Review by thehejik — APPROVED** [review]
It was tested manually on s390x hardware, everything works as expected.

File: scripts/build-binary
Comment (dweomer) [inline scripts/build-binary:27]: We should retain the usage of `${IMAGE_BUILD_VERSION}` should we not?

File: scripts/build-s390x
Comment (dweomer) [inline scripts/build-s390x:12]: is this just a copy of scripts/build? with all of these other changes is `make build` on an s390x host not sufficient?

File: scripts/build-binary
Comment (raulcabello) [reply scripts/build-binary:27]: I changed it because I saw in master was like this https://github.com/rancher/rke2/blob/2faa95b6a8b3a3abe12fb90c4529f7aba29803da/scripts/build-binary#L27

File: scripts/build-s390x
Comment (raulcabello) [reply scripts/build-s390x:12]: it is not a copy. We don't want to build window images and we don't support airgap on s390x

File: scripts/build-binary
Comment (brandond) [reply scripts/build-binary:27]: It made sense to have a variable when it looked like we were going to build all the images at the same time, but now with things being as ad-hoc as they are - many images have different build tags -  it's not really getting used as I thought it would. I'm OK with dropping it.

File: scripts/build-s390x
Comment (brandond) [reply scripts/build-s390x:12]: Do we have a design document that captures the omission of airgap support for this architecture? That seems like an important thing to casually omit.

**Review by brandond — CHANGES_REQUESTED** [review]
LGTM, once the docs are updated to cover lack of airgap support for s390x (if that is OK as per Product).

**Review by briandowns — APPROVED** [review]
LGTM but would like to see a docs update like @brandond mentioned.

**Comment by agracey** [comment]
PM here. For now, the lack of airgap is expected. We will revisit in a few months after a few POCs

**Comment by agracey** [comment]
To clarify. We need air gap for our GA which we are hoping to achieve in March. 

What is the amount of work required to get it now?

File: scripts/build-s390x
Comment (agracey) [reply scripts/build-s390x:12]: What is missing here to provide air gap? I was under the impression that air gap was effectively pulling the images down locally and "side-loading" them into the server(s) or a local registry. Is there more to it than that?

Comment (brandond) [reply scripts/build-s390x:12]: I would probably recommend
* Putting the windows bits of `./scripts/build` behind an architecture check, or perhaps modifying the various `scripts/*-windows-*` scripts to exit early on non-amd64 architectures so that they become a no-op.
* Modifying `scripts/build-images` to skip building airgap tarballs for the optional components on non-amd64 architectures.
Everything else should already work, as far as I can tell?

Comment (raulcabello) [reply scripts/build-s390x:12]: Yes, that should work. Thanks!

**Comment by raulcabello** [comment]
@agracey it shouldn’t take long to add airgap for s390x. But I won’t be able to do it until next Tuesday, and I think code freeze is next week.
The work to do it in a month should be the same as doing it now. The only extra thing we would need to do is to update the docs to say we don’t support airgap for s390x. Or mention somewhere airgap is not available for the technical preview.

File: Makefile
Comment (brandond) [inline Makefile:61]: I think this can go away now that the script has been removed

File: scripts/build
Comment (brandond) [inline scripts/build:18]: I am not confident that we can reorder these like this without breaking some of the build dependencies in ways that will not cause tests or the pipeline to fail, but may cause some artifacts to not be published. I'm also curious why we wouldn't want dev-runtime-image and build-image-test on s390x.

Can you leave the order here as-is, and add arch checks to the Windows scripts that just immediately `exit 0` on the wrong arch?

**Review by brandond — CHANGES_REQUESTED** [review]

File: Dockerfile
Comment (brandond) [inline Dockerfile:120]: Why the extra newline?

I think we also need to bump the image in the rke2-kube-proxy chart, and then bump the chart version up above.

**Review by brandond — CHANGES_REQUESTED** [review]

File: Dockerfile
Comment (raulcabello) [reply Dockerfile:120]: https://github.com/rancher/rke2-charts/pull/196 just bumped the chart, can you have a look?

File: Makefile
Comment (raulcabello) [reply Makefile]: done in https://github.com/rancher/rke2/pull/2171/commits/133611b8d82876240448ae1357cdf8a86f60fe60

File: scripts/build
Comment (raulcabello) [reply scripts/build]: done in https://github.com/rancher/rke2/pull/2171/commits/75921506838b7dd92924ea953168af0522501cc8

File: Dockerfile
Comment (raulcabello) [reply Dockerfile]: bump and remove line https://github.com/rancher/rke2/pull/2171/commits/c229aa7a368714892b692352324a38eb0e94a316

Comment (raulcabello) [reply Dockerfile]: ah build failed because it could not fin chart https://drone-pr.rancher.io/rancher/rke2/2658/1/2 
Should pass after https://github.com/rancher/rke2-charts/pull/196 is merged

File: scripts/build-images
Comment (brandond) [inline scripts/build-images:19]: if there's nothing left using `IMAGE_BUILD_VERSION` can you also remove it from version.sh to avoid confusion?

**Review by brandond — APPROVED** [review]
LGTM, with one nit about cleaning up the now-unused variable.

Make sure this gets squashed before merging?

File: scripts/build-images
Comment (raulcabello) [reply scripts/build-images:19]: It's not being used. Done in https://github.com/rancher/rke2/pull/2171/commits/68303a64afce935f5c5f0dd387c7c22d341587a1

**Comment by raulcabello** [comment]
@brandond Can you squash it when merging in github? If not I can squash the commits in my local and do a force push. 
Do we need another approval? I don't have permission to merge, so someone else will need to do it.

**Pull request merged**
